polyphemus/lib/client/jsx/etl/etl-config.tsx

```diff
@@ -86,74 +82,152 @@ const useStyles = makeStyles( theme => ({
   }
 }));
 
-const EtlConfig = ({project_name, etl,name,config,status,secrets,params,output,run_interval,ran_at,job,onUpdate}: Etl & {job:Job|undefined,onUpdate:Function}) => {
-  const [ mode, setMode ] = useState<string | null>(null);
-  const [ error, setError ] = useState('');
-  const toggleMode = (m:string) => mode == m ? setMode(null) : setMode(m);
+const EtlConfig = ({
+  project_name,
+  etl,
+  name,
+  config,
+  status,
+  secrets,
+  params,
+  output,
+  run_interval,
+  ran_at,
+  job,
+  onUpdate
+}: Etl & {job: Job | undefined; onUpdate: Function}) => {
+  const [mode, setMode] = useState<string | null>(null);
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState('');
+
+  const clearMessages = useCallback(() => {
+    setMessage('');
+    setError('');
+  }, []);
 
-  const classes:any = useStyles();
+  const toggleMode = (m: string) => {
+    mode == m ? setMode(null) : setMode(m);
+    clearMessages();
+  };
 
-  const postUpdate = (update:any) => json_post(
-    `/api/etl/${project_name}/update/${name}`, update
-  ).then( etl => onUpdate(etl) ).catch(
-    r => r.then(({error}:{error:string}) => setError(error))
+  const classes: any = useStyles();
+
+  const [_, postUpdate] = useAsyncWork(
+    function postUpdate(update: any) {
+      clearMessages();
+      return json_post(`/api/etl/${project_name}/update/${name}`, update)
+        .then((etl) => {
+          onUpdate(etl);
+          setMessage('Saved!');
```

Comment:
This is there once you save but doesn't go away if you reset or change the state, so some confusion remains. Could the buttons be grayed instead if there are no changes?

Reply:
Oh, good catch! I only tested that it goes away if you navigate to a different panel. Thanks, will update how that works.

Reply:
Hm...grayed from the last saved state? Probably, will have to see how to recalculate that.